tests/RecordTests/FunctionalTest/k4a_cpp_ft.cpp

```diff
@@ -0,0 +1,453 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <k4arecord/record.hpp>
+#include <k4arecord/playback.hpp>
+#include <utcommon.h>
+
+#include <gtest/gtest.h>
+
+using namespace k4a;
+
+int main(int argc, char **argv)
+{
+    return k4a_test_common_main(argc, argv);
+}
+
+class k4a_cpp_ft : public ::testing::Test
+{
+public:
+    virtual void SetUp() {}
+
+    virtual void TearDown() {}
+};
+
+TEST_F(k4a_cpp_ft, k4a)
+{
+    device kinect = device::open(0);
+
+    ASSERT_THROW(device kinect2 = device::open(0), error);
+    kinect.close();
+    kinect = kinect.open(0);
+
+    {
+        ASSERT_LE((uint32_t)1, device::get_installed_count());
+    }
+
+    {
+        k4a_hardware_version_t version = kinect.get_version();
+        ASSERT_GE(version.rgb.major, (uint32_t)1);
+        ASSERT_GE(version.rgb.minor, (uint32_t)6);
```

Comment:
This will break if we ever upgrade to a 2.0 firmware.
Maybe that's not important and we can just update the test then.

Reply:
Yea, we will worry about this one when there is a 2.0 or greater.